scalableminds/webknossos
# Add Tags to Explorative Tracing Dashboard

### Mailable description of changes:
- Added a tagging system to tracings
- Tags can be easily searched and filtered for explorative tracing in the dashboard
- By default, the dataset name and "skeleton" / "volume" are added automatically as tags
- The "Archive All" button respects the current search / tag filtering now. Only the currently visible tracings are archived. A confirmation modal will display the number of affected tracings and ask for confirmation.

### Steps to test:
- Go to dashboard --> explorative tracings
- Add new custom tags by clicking on "plus icon".
- Reload page to see if they were saved
- Filter the list by clicking on several tags
- Reload page to see if the tag selection was persisted through localStorage
- Use the search bar to filter the list even further
- Use the search bar to add new "search tags". Type a search query and hit enter.
- Check that the list is filtered using an AND semantic

### Issues:
- fixes #1971 

@philippotto Please review the frontend changes.
@rschwanhold Please review the backend changes.

See  https://discuss.webknossos.org/t/improvements-to-the-explorative-annotations-dashboard-tab/408/14

------
